openapi-core/src/main/java/io/ballerina/openapi/core/GeneratorUtils.java

```diff
@@ -1037,4 +1044,51 @@ private static SemanticModel getSemanticModel(Path clientPath) throws ProjectExc
             throw new ProjectException(e.getMessage());
         }
     }
+
+    public static String getOpenAPIType(Schema<?> schema) {
+        if (schema.getTypes() != null && !schema.getTypes().isEmpty()) {
+            for (String type : schema.getTypes()) {
+                // this returns the first non-null type in the list
```

Comment:
Shall we add a todo here (and create an issue) to handle the other types, incase there will be multiple types?

Reply:
Added